niap-cc/Permissions/Companion/app/src/main/java/com/android/certifications/niap/permissions/companion/MainActivity.java

```diff
@@ -255,7 +263,7 @@ public void onReceive(Context context, Intent intent) {
 
         registerReceiver(receiver, new IntentFilter(ACCESS_LOCATION_ACTION));
         Intent intent = new Intent(ACCESS_LOCATION_ACTION);
-        PendingIntent pendingIntent = PendingIntent.getBroadcast(this, 0, intent, 0);
+        PendingIntent pendingIntent = PendingIntent.getBroadcast(this, 0, intent, FLAG_MUTABLE);
```

Comment:
Just out of curiosity, is `FLAG_MUTABLE` used here because the locationClient will need to modify the broadcast?

Reply:
The flag should be assigned after Target SDK31, So I had assigned it. maytbe it's more appropriate to assign FLAG_IMMUTABLE. Let me check

Reply:
Hi, as I checked the value should be FLAG_IMMUTABLE, otherwise the intent does not return value and application would be stacked. as you said it looks like the location client can not change the value in that case.

Reply:
Sounds good, thanks for confirming!